apps/web/vibes/soul/primitives/dynamic-form/schema.ts

```diff
@@ -0,0 +1,234 @@
+
+type SelectField = {
+  type: 'select';
+  options: { label: string; value: string }[];
+  defaultValue?: string;
+} & FormField;
+
+type CheckboxField = {
+  type: 'checkbox';
+  defaultValue?: string;
+} & FormField;
+
+type CheckboxGroupField = {
+  type: 'checkbox-group';
+  options: { label: string; value: string }[];
+  defaultValue?: string[];
+} & FormField;
+
+type NumberInputField = {
+  type: 'number';
+  defaultValue?: string;
+  min?: number;
+  max?: number;
+  step?: number;
+} & FormField;
+
+type TextInputField = {
+  type: 'text';
+  defaultValue?: string;
+} & FormField;
+
+type EmailInputField = {
+  type: 'email';
+  defaultValue?: string;
+} & FormField;
+
+type TextAreaField = {
+  type: 'textarea';
+  defaultValue?: string;
+} & FormField;
+
+type DateField = {
+  type: 'date';
+  defaultValue?: string;
+  minDate?: string;
+  maxDate?: string;
+} & FormField;
+
+type SwatchRadioFieldOption =
+  | {
+      type: 'color';
+      value: string;
+      label: string;
+      color: string;
+      disabled?: boolean;
+    }
+  | {
+      type: 'image';
+      value: string;
+      label: string;
+      image: { src: string; alt: string };
+      disabled?: boolean;
+    };
+
+type SwatchRadioField = {
+  type: 'swatch-radio-group';
+  defaultValue?: string;
+  options: SwatchRadioFieldOption[];
+} & FormField;
+
+type CardRadioField = {
+  type: 'card-radio-group';
+  defaultValue?: string;
+  options: {
+    value: string;
+    label: string;
+    image: { src: string; alt: string };
+    disabled?: boolean;
+  }[];
+} & FormField;
+
+type ButtonRadioField = {
+  type: 'button-radio-group';
+  defaultValue?: string;
+  pattern?: string;
+  options: {
+    value: string;
+    label: string;
+    disabled?: boolean;
+  }[];
+} & FormField;
+
+type PasswordField = {
+  type: 'password';
+} & FormField;
+
+type ConfirmPasswordField = {
+  type: 'confirm-password';
+} & FormField;
+
+export type Field =
+  | RadioField
+  | CheckboxField
+  | CheckboxGroupField
+  | NumberInputField
+  | TextInputField
+  | TextAreaField
+  | DateField
+  | SwatchRadioField
+  | CardRadioField
+  | ButtonRadioField
+  | SelectField
+  | PasswordField
+  | ConfirmPasswordField
+  | EmailInputField;
+
+export type FieldGroup<F> = F[];
+
+export type SchemaRawShape = Record<
+  string,
+  | z.ZodString
+  | z.ZodOptional<z.ZodString>
+  | z.ZodNumber
+  | z.ZodOptional<z.ZodNumber>
+  | z.ZodArray<z.ZodString>
+  | z.ZodOptional<z.ZodArray<z.ZodString>>
+>;
+
+function getFieldSchema(field: Field) {
+  let fieldSchema:
+    | z.ZodString
+    | z.ZodNumber
+    | z.ZodOptional<z.ZodString>
+    | z.ZodOptional<z.ZodNumber>
+    | z.ZodArray<z.ZodString, 'atleastone' | 'many'>
+    | z.ZodOptional<z.ZodArray<z.ZodString, 'atleastone' | 'many'>>;
+
+  switch (field.type) {
+    case 'number':
+      fieldSchema = z.number();
+
+      if (field.min != null) fieldSchema.min(field.min);
+      if (field.max != null) fieldSchema.max(field.max);
```

Comment:
Both of these seem to be no-ops since we're not reassigning `fieldSchema`

Reply:
Ah good catch